packages/spatial/src/renderer/WebGLRendererSystem.test.tsx

```diff
@@ -0,0 +1,193 @@
+    getMutableComponent(rootEntity, RendererComponent).set(new MockEngineRenderer())
+    const rendererComp = getMutableComponent(rootEntity, RendererComponent)
+    rendererComp.canvas.set(mockCanvas())
+    setComponent(rootEntity, BackgroundComponent, new Color(0x000000))
+
+    setComponent(rootEntity, EnvironmentMapComponent, new Texture())
+    setComponent(rootEntity, FogSettingsComponent, { type: FogType.Height })
+
+    invisibleEntity = createEntity()
+    setComponent(invisibleEntity, UUIDComponent, MathUtils.generateUUID() as EntityUUID)
+    setComponent(invisibleEntity, GroupComponent)
+    const invisibleObject3d = setComponent(invisibleEntity, Object3DComponent, new Group())
+    addObjectToGroup(invisibleEntity, invisibleObject3d)
+    setComponent(invisibleEntity, EntityTreeComponent)
+    setComponent(rootEntity, SceneComponent, { children: [invisibleEntity] })
+
+    visibleEntity = createEntity()
+    setComponent(visibleEntity, UUIDComponent, MathUtils.generateUUID() as EntityUUID)
+    setComponent(visibleEntity, VisibleComponent)
+    const visibleObject3d = setComponent(visibleEntity, Object3DComponent, new Group())
+    addObjectToGroup(visibleEntity, visibleObject3d)
+    setComponent(visibleEntity, GroupComponent)
+    setComponent(visibleEntity, EntityTreeComponent)
+    setComponent(rootEntity, SceneComponent, { children: [visibleEntity] })
+
+    nestedInvisibleEntity = createEntity()
+    setComponent(nestedInvisibleEntity, UUIDComponent, MathUtils.generateUUID() as EntityUUID)
+    setComponent(nestedInvisibleEntity, GroupComponent)
+    const nestedInvisibleObject3d = setComponent(nestedInvisibleEntity, Object3DComponent, new Group())
+    addObjectToGroup(nestedInvisibleEntity, nestedInvisibleObject3d)
+    setComponent(nestedInvisibleEntity, EntityTreeComponent)
+    setComponent(visibleEntity, SceneComponent, { children: [nestedInvisibleEntity] })
+
+    nestedVisibleEntity = createEntity()
+    setComponent(nestedVisibleEntity, UUIDComponent, MathUtils.generateUUID() as EntityUUID)
+    setComponent(nestedVisibleEntity, VisibleComponent)
+    const nestedVisibleObject3d = setComponent(nestedVisibleEntity, Object3DComponent, new Group())
+    addObjectToGroup(nestedVisibleEntity, nestedVisibleObject3d)
+    setComponent(nestedVisibleEntity, GroupComponent)
+    setComponent(nestedVisibleEntity, EntityTreeComponent)
+    setComponent(invisibleEntity, SceneComponent, { children: [nestedVisibleEntity] })
+
+    //override addpass to test data without dependency on Browser
+    let addPassCount = 0
+    EffectComposer.prototype.addPass = () => {
+      addPassCount++
+    }
+  })
+
+  afterEach(() => {
+    return destroyEngine()
+  })
+
+  it('test', async () => {
+    const { background, environment, fog, children } = getSceneParameters([rootEntity])
+    SystemDefinitions.get(WebGLRendererSystem)?.execute()
+    globalThis._scene
+    console.log('test')
```

Comment:
extra console log here and the name of the test is not super descriptive

Reply:
corrected